lib/vsc/filesystem/quota/ktools.py

```diff
@@ -0,0 +1,382 @@
+                    cl = cl.user
+                elif self.kind == UsageType.FilesetUsage:
+                    logging.debug("Pushing vo payload to account web app: %s", payload)
+                    cl = cl.vo
+                else:
+                    logging.error("Unknown quota kind, not pushing any quota to the account page")
+                    return
+                cl.size.put(body=payload)  # if all is well, there's nothing returned except (200, empty string)
+            except Exception:
+                logging.error("Could not store quota info in account web app")
+                raise
+
+
+def map_uids_to_names():
+    """Determine the mapping between user ids and user names."""
+    ul = pwd.getpwall()
+    d = {}
+    for u in ul:
+        d[u[2]] = u[0]
+    return d
+
+
+def determine_grace_period(grace_string):
+    grace = GPFS_GRACE_REGEX.search(grace_string)
+    nograce = GPFS_NOGRACE_REGEX.search(grace_string)
+
+    if nograce:
+        expired = (False, None)
+    elif grace:
+        grace = grace.groupdict()
+        grace_time = 0
+        if grace['days']:
+            grace_time = int(grace['days']) * 86400
+        elif grace['hours']:
+            grace_time = int(grace['hours']) * 3600
+        elif grace['minutes']:
+            grace_time = int(grace['minutes']) * 60
+        elif grace['expired']:
+            grace_time = 0
+        else:
+            logging.error("Unprocessed grace groupdict %s (from string %s).",
+                          grace, grace_string)
+            raise QuotaException("Cannot process grace time string")
+        expired = (True, grace_time)
+    else:
+        logging.error("Unknown grace string %s.", grace_string)
+        raise QuotaException("Cannot process grace information (%s)" % grace_string)
+
+    return expired
+
+
+def process_user_quota(vsc, pusher, quota, user_map, institute=GENT):
+
+    logging.info("Logging user quota to account page")
+    logging.debug("Considering the following quota items for pushing: %s", quota)
+
+    for (user_id, fileset, quota) in quota:
+
+        user_institute = vsc.user_id_to_institute(int(user_id))
+        if user_institute != institute:
+            continue
+
+        user_name = user_map.get(int(user_id), None)
+        if not user_name:
+            continue
+
+        fileset_re = '^(vsc[1-4]|%s|%s)' % (
+            VO_PREFIX_BY_SITE[institute],
+            VO_SHARED_PREFIX_BY_SITE[institute])
+
+        if re.search(fileset_re, fileset):
+            pusher.push_quota(user_name, fileset, quota)
+
+
+def process_fileset_quota(pusher, quota, institute=GENT):
+
+    logging.info("Logging fileset quota to account page")
+    logging.debug("Considering the following quota items for pushing: %s", quota)
+
+    for (fileset, quota) in quota:
+
+        if not fileset.startswith(VO_PREFIX_BY_SITE[institute]):
+            continue
+
+        if fileset.startswith(VO_SHARED_PREFIX_BY_SITE[institute]):
```

Comment:
this is only reached because VO_PREFIX_BY_SITE has the same start as VO_SHARED_PREFIX_BY_SITE ...